### DISCUSSION TITLE
BEGINNERS' CORNER

### DISCUSSION THREAD

**Discussion by Sorcerer13** [pybricks/support#1300] [General]
I used to be a professional programmer (I'm now retired), and some fool gave me a Lego Spike kit.
I've followed a few build projects, and then got Anton's Mindstorm board for a remote control option, which led me to Pybricks code.
I've seen 'https://docs.pybricks.com/en/latest/index.html' as documentation for Pybricks, but there are some basic (very!!!) questions left unanswered.
Primarily these are
1) How do I introduce comments as a) comment at the end of a statement, b) single line comment, c) multiple line comment, and
2) How do I split Pybricks code statements across multiple lines, for example:-
Invoke_something(Parameter1,
                              Parameter 2,
                              Parameter 3)
It's all probably blindingly obvious, but I still can't find an Idiot's Guide which covers such basic topics such as the structure of a statement etc.
Can any kind soul help?

**Comment by Sorcerer13** [comment]
In the above example Parameters 1 to 3 should be aligned vertically, starting around column 18.
Sorry!

**Comment by laurensvalk** [comment]
Thanks for your note. You can find various examples with comments on the [projects site](https://pybricks.com/projects/sets/technic/42131-cat-bulldozer/powered-up-remote/) as well as throughout the documentation you've linked to. Pybricks is based on Python and follows all the same syntax rules, including comments.

We don't currently have a complete beginner's guide yet. However, we'll soon release a visual code editor which will help assemble code visually, so you get the correct syntax automatically. We're planning to add a comment block, too!

**Comment by jmancoder** [comment]
The basic syntax is largely the same as Python, and there are plenty of basic Python tutorials available for beginners. Therefore, there's really no need to reinvent the wheel in the Pybricks documentation.

You can make comments by prefacing a line with #. Here are a few examples:
```
print("Hello World!") # This is a comment after a statement
# This is a single-line comment
# This is kind of a multi-line comment
"""
This is a slightly more hacky way of
making a multi-line comment.
"""
```

As for splitting statements into multiple lines, you can simply indent each sub-line. For example:
```
print("Hello",
    "Wonderful",
    "World")
```

**Comment by Sorcerer13** [comment]
Thanks for the update, jmancoder.
Being severely sight impaired I find Python editors a little awkward to deal with, as they seem to do a lot of 'helpful' things like auto indent, which can be a pain if you use a 600x800 resolution and you find yourself scrolling horizontally quite a lot, which is why I enquired about continuation lines.
I guess I'll just have to persevere (or buy a much much larger TV to use as a screen!

**Reply by BertLindeman** [reply]
@Sorcerer13

In discussion #1367 David indicated how to verify the possibilities in the pybricks editor.
An option for you might be to change the tab indent from 4 (the default) to 2 so you have to scroll just a bit less?

Bert

**Reply by Sorcerer13** [reply]
Thanks for the suggestion, but how? I've invoked Pybricks (online via https://code.pybricks.com/) and written some code, but I cant see a 'Settings' option or anything else whick looks as if it could lead to a Settings area.
Is it in a Pybricks application which I could download?

**Reply by laurensvalk** [reply]
For app settings, you can click the gear icon on the left. For additional editor settings, hit F1 and search for an option.

If you use 2 spaces for indentation, you can tell the editor to pick this up and use it going forward:

![image](https://github.com/pybricks/support/assets/12326241/60361450-e25f-431e-a1bb-f7efe668a41b)

**Comment by Sorcerer13** [comment]
OK - I think I found it via:-
 In Pybricks, create a new project.
Type something like
```python
  #Comment with two leading spaces before the hash.
```
Keep cursor on that line, hit F1, enter "Indent" in the search box, then from the options select "Detect Indentation from Content".

Thanks!
